samza-core/src/test/java/org/apache/samza/clustermanager/TestContainerPlacementActions.java

```diff
@@ -558,14 +548,19 @@ public Void answer(InvocationOnMock invocation) {
   public void testContainerPlacementsForJobRunningInDegradedState() throws Exception {
     // Set failure after retries to false to enable job running in degraded state
     config = new MapConfig(configVals, getConfigWithHostAffinityAndRetries(true, 1, false));
-    state = new SamzaApplicationState(getJobModelManagerWithHostAffinity(ImmutableMap.of("0", "host-1", "1", "host-2")));
+    state = new SamzaApplicationState(JobModelManagerTestUtil.getJobModelManager(getConfig(), 2, this.server));
     callback = mock(ClusterResourceManager.Callback.class);
     MockClusterResourceManager clusterResourceManager = new MockClusterResourceManager(callback, state);
     ClusterManagerConfig clusterManagerConfig = new ClusterManagerConfig(config);
-    containerManager = spy(new ContainerManager(containerPlacementMetadataStore, state, clusterResourceManager, true, false));
+    LocalityManager mockLocalityManager = mock(LocalityManager.class);
```

Comment:
redundant, isn't this already done in setup?

Reply:
extracted the locality manager to instance variable to reuse them in the tests.